awc/src/request.rs

```diff
@@ -382,6 +382,25 @@ impl ClientRequest {
         }
     }
 
+    /// Sets the query part of the request
+    pub fn query<T: Serialize>(mut self, query: &T) -> Self {
+        let mut parts = self.head.uri.clone().into_parts();
+
+        if let Some(path_and_query) = parts.path_and_query {
+            if let Ok(query) = serde_urlencoded::to_string(query) {
```

Comment:
As the implementation is right now, it would silently fail if the serde_urlencoded fails to serialize the `query` argument, I believe that the `self.err` of the `ClientRequest` should be set, but didn't manage to find a way to create a `http::Error` from serde's or from `actix_http::Error`. Any suggestions on how to handle this?

Reply:
I think you should be able to to do `actix_http::Error::from(e)` for serde errors. you may need to add new request's attribute for serde_urlencoded errors, then [prep_for_sending](https://github.com/actix/actix-web/blob/master/awc/src/request.rs#L492) should accommodate that error.

Reply:
I can generate a `actix_http::Error` from a serde_urlencoded without any problems, but I didn't have a way to store that on the `awc::request::ClientRequest`, so I added a `actix_error` to that is handled by the prep_for_sending as well. But to make the conversion from `actix_http::Error` to a `actix_http::client::FreezeRequestError` I had to add a enum item to the `actix_http::client::FreezeRequestError`.

With this approach a update to the actix_http would have to come first, I've been testing setting the path from actix_http to '../actix_http', but I didn't add this to the PR.

Maybe a better way to handle this would be change the err on `awc::request::ClientRequest` itself to be a `actix_http::Error` insted of having two errors fields, and since we have a From `actix_http::Error` to `http::Error`, doing this should lead to any problems other than having to change a feel types and adding a from here and there.

What do you think? Should I keep two 'err' fields on ClientRequest, or should I change the err type and have only one?

Reply:
I don't like idea with actix_http::Error in general, it is fine when it get returned back to framework (ie into actix-web), but i think it is not good to expose it to user. let's make `query()` method return `Result<Self, ..>`